src/compas_rhino/geometry/brep/brep.py

```diff
@@ -0,0 +1,248 @@
+from compas.geometry import Frame
+from compas.geometry import Brep
+from compas.geometry import BrepInvalidError
+from compas.geometry import BrepTrimmingError
+
+from compas_rhino.conversions import box_to_rhino
+from compas_rhino.conversions import point_to_rhino
+from compas_rhino.conversions import xform_to_rhino
+from compas_rhino.conversions import frame_to_rhino
+
+import Rhino
+
+from .face import RhinoBrepFace
+from .edge import RhinoBrepEdge
+from .vertex import RhinoBrepVertex
+from .loop import RhinoBrepLoop
+
+
+TOLERANCE = 1e-6
+
+
+class RhinoBrep(Brep):
+    """
+    Rhino Brep backend class.
+    Wraps around and allows serialization and de-serialization of a `Rhino.Geometry.Brep`
+    """
+
+    # this makes de-serialization backend-agnostic.
+    # The deserializing type is determined by plugin availability when de-serializing
+    # regardless of the context available when serializing.
+    __class__ = Brep
+
+    def __new__(cls, *args, **kwargs):
+        # This breaks the endless recursion when calling `compas.geometry.Brep()` and allows
+        # having Brep here as the parent class. Otherwise RhinoBrep() calls Brep.__new__()
+        # which calls RhinoBrep() and so on...
+        return object.__new__(cls, *args, **kwargs)
```

Comment:
i don't think this is correct. cf. the Curve and Surface constructors...

Reply:
We discussed this and decided to leave it for the time being. 

For the sake of documentation, this is to allow:
1. having `Brep` as a super class of `RhinoBrep`
2. creating an empty e.g. `RhinoBrep` using just `Brep()`


The way it's done in e.g. `Surface` makes 1 possible but not 2.